src/scanner.c

```diff
@@ -2122,29 +2122,28 @@ void GetChar ( void )
   GET_CHAR();
 
   /* handle escape equences                                              */
-  if ( *TLS(In) == '\\' ) {
-    GET_CHAR();
-    TLS(Value)[0] = GetEscapedChar();
-  }
-  else if ( *TLS(In) == '\n' ) {
-    SyntaxError("Newline not allowed in character literal");
-  }
-  /* put normal chars into 'TLS(Value)'                                  */
-  else {
-    TLS(Value)[0] = *TLS(In);
-  }
-
-  /* read the next character                                             */
-  GET_CHAR();
-
-  /* check for terminating single quote                                  */
-  if ( *TLS(In) != '\'' )
-    SyntaxError("Missing single quote in character constant");
+  if ( *TLS(In) == '\n' ) {
+    SyntaxError("Character literal must not include <newline>");
+  } else {
```

Comment:
This commit has the description "Refactor GetChar()", but for me "refactor" suggests that there is no functional difference to the previous code. But that does not seem to be the case for this refactoring. For example, old and new code differ on how they handle a newline (both give a syntax error, but a different one). So at the very least, I find the commit message misleading.